app/src/main/java/com/android/bookswap/ui/map/FilterMap.kt

```diff
@@ -89,18 +90,22 @@ fun FilterMapScreen(navigationActions: NavigationActions, selectedFilters: Mutab
         LazyColumn(contentPadding = paddingValues, modifier = Modifier.fillMaxSize()) {
           item {
             ButtonBlock(
-                BookGenres.values().map {
-                  it.toString().lowercase().replaceFirstChar { c -> c.uppercase() }
-                },
-                selectedFiltersTemp)
+                BookGenres.values().map { it.Genre }, selectedFiltersGenres.map { it.Genre }) {
+                    newSelection ->
+                  bookFilter.setGenres(newSelection)
+                }
           }
           item {
```

Comment:
It would be nice to have some comments here as well, I find myself reading things anew more than I would normally wish, and while it seems to be fine, it does show that perhaps some comments might aid in its lecture.